java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java

```diff
@@ -130,7 +130,7 @@ private void handleSuppressWarning(List<AnnotationTree> annotationTrees, Tree tr
     List<String> rules = Lists.newArrayList();
     for (AnnotationTree annotationTree : annotationTrees) {
       if (isSuppressWarningsAnnotation(annotationTree)) {
-        startLine = ((JavaTree) annotationTree).getLine();
+        startLine = startLineIncludingTrivia(tree);
```

Comment:
Yes, the start line computation has to be modified to account for the javadoc, why is that an issue? The example you gave works as it should. Perhaps I didn't understood the comment? What kind of behavior you expect?